yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnAllocator.scala

```diff
@@ -423,6 +424,7 @@ private[yarn] class YarnAllocator(
     for (completedContainer <- completedContainers) {
       val containerId = completedContainer.getContainerId
       val alreadyReleased = releasedContainers.remove(containerId)
+      var completedContainerReason : String = ""
```

Comment:
this is kinda subjective, but its often nice to use a `val` which gets assigned as the result of the conditionals.  That way its clear that the value is fixed at the end.  The downside is that sometimes it makes the assignment a bit harder to follow eg. if there is a lot of nesting.  this case is kinda borderline IMO.  in any case, the code would look like:

```scala
val completedContainerReason: Option[String] = if (!alreadyReleased) {
  ...
  if (completedContainer.getExitStatus = ContainerExistStatus.PREEMPTED) {
     ...
     Some(s"Container preempted: $containerId") // this is the "result" of the if
  } else if ... {
   ...
   
  }
} else {
  None
}
```